src/celeritas/global/ActionLauncher.device.hh

```diff
@@ -94,7 +91,7 @@ class ActionLauncher
             CELER_DEVICE_PREFIX(Stream_t)
             stream = celeritas::device().stream(stream_id).get();
             auto config = calc_launch_params_(threads.size());
-            launch_action_impl<F>
+            detail::launch_action_impl<F>
                 <<<config.blocks_per_grid, config.threads_per_block, 0, stream>>>(
                     threads, call_thread);
```

Comment:
there's a bug if we specify a `__launch_bounds__` with max_threads_per_blocks smaller than the computed threads_per_block, some tracks will never get processed.

Reply:
As the CUDA documentation says, that behavior is expected: 

> A kernel will fail to launch if it is executed with more threads per block than its launch bound maxThreadsPerBlock.

we probably want to allow different block size though, for example I was doing a profiling of the pre-step kernel and changing the block size to 224 threads/block would increase the occupancy from 38% to 44%.

I'll have to think what would be the best way to prevent specifying a launch bounds smaller than what is used to launch the kernel.

Reply:
The easiest work around I found is to pass the max_block_size, if it exists, to the KernelParamsCalculator which then adjust the number of blocks accordingly.

Reply:
Nice fix. Since `register_kernel` actually fills a `KernelAttributes` which contains the maximum block size, we could also add a validation check there.